samples/azure/azure-service-fabric-routing/Core_6/CandidateVoteCount/ServiceEventSource.cs

```diff
@@ -0,0 +1,163 @@
+        public static class Keywords
+        {
+            public const EventKeywords Requests = (EventKeywords)0x1L;
+            public const EventKeywords ServiceInitialization = (EventKeywords)0x2L;
+        }
+
+        // Define an instance method for each event you want to record and apply an [Event] attribute to it.
+        // The method name is the name of the event.
+        // Pass any parameters you want to record with the event (only primitive integer types, DateTime, Guid & string are allowed).
+        // Each event method implementation should check whether the event source is enabled, and if it is, call WriteEvent() method to raise the event.
+        // The number and types of arguments passed to every event method must exactly match what is passed to WriteEvent().
+        // Put [NonEvent] attribute on all methods that do not define an event.
+        // For more information see https://msdn.microsoft.com/en-us/library/system.diagnostics.tracing.eventsource.aspx
+
+        [NonEvent]
+        public void Message(string message, params object[] args)
+        {
+            if (this.IsEnabled())
+            {
+                string finalMessage = string.Format(message, args);
+                Message(finalMessage);
+            }
+        }
+
+        private const int MessageEventId = 1;
+        [Event(MessageEventId, Level = EventLevel.Informational, Message = "{0}")]
+        public void Message(string message)
+        {
+            if (this.IsEnabled())
+            {
+                WriteEvent(MessageEventId, message);
+            }
+        }
+
+        [NonEvent]
+        public void ServiceMessage(StatefulServiceContext serviceContext, string message, params object[] args)
+        {
+            if (this.IsEnabled())
+            {
+                string finalMessage = string.Format(message, args);
+                ServiceMessage(
+                    serviceContext.ServiceName.ToString(),
+                    serviceContext.ServiceTypeName,
+                    serviceContext.ReplicaId,
+                    serviceContext.PartitionId,
+                    serviceContext.CodePackageActivationContext.ApplicationName,
+                    serviceContext.CodePackageActivationContext.ApplicationTypeName,
+                    serviceContext.NodeContext.NodeName,
+                    finalMessage);
+            }
+        }
+
+        // For very high-frequency events it might be advantageous to raise events using WriteEventCore API.
+        // This results in more efficient parameter handling, but requires explicit allocation of EventData structure and unsafe code.
+        // To enable this code path, define UNSAFE conditional compilation symbol and turn on unsafe code support in project properties.
+        private const int ServiceMessageEventId = 2;
+        [Event(ServiceMessageEventId, Level = EventLevel.Informational, Message = "{7}")]
+        private
+#if UNSAFE
```

Comment:
i know this was generated by the template. but if it is not required by us can it be removed?